src/pages/Integrations/Create/CustomComponents/FinalStep.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import CreatedStep from './CreatedStep';
+import FailedStep from './FailedStep';
+import {
+  Button,
+  EmptyState,
+  EmptyStateActions,
+  EmptyStateFooter,
+  EmptyStateHeader,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Spinner,
+} from '@patternfly/react-core';
+
+export type IntegrationsData = {
+  url: string;
+  type: string;
+  sub_type?: string;
+  name: string;
+  secret_token?: string;
+  isEdit?: boolean;
+  template?: {
+    id: string;
+  };
+  event_type_id: [];
+  bundle_name: string;
+};
+
+interface ProgressProps {
+  data: IntegrationsData;
+  onCancel: () => void;
+}
+
+export const FinalStep: React.FunctionComponent<ProgressProps> = ({
+  data,
+  onCancel,
+}) => {
+  const [isFinished, setIsFinished] = React.useState(false);
+  const [hasError, setHasError] = React.useState(false);
+
+  const integrationsUrl = '/api/integrations/v1.0/endpoints';
+  const behaviorGroupUrl = `/api/notifications/v1.0/notifications/behaviorGroups`;
+
+  React.useEffect(() => {
+    const createAction = async () => {
+      try {
+        const result = await (
+          await fetch(
+            `${integrationsUrl}${data.isEdit ? `/${data.template?.id}` : ''}`,
+            {
+              method: data.isEdit ? 'PUT' : 'POST',
+              headers: {
+                'Content-Type': 'application/json;charset=UTF-8',
+              },
+              body: JSON.stringify({
+                name: data.name,
+                enabled: true,
+                type: data.type,
+                ...(data.sub_type && { sub_type: data.sub_type }),
+                description: '',
+                properties: {
+                  method: 'POST',
+                  url: data.url,
+                  disable_ssl_verification: false,
+                  secret_token: data.secret_token,
+                },
+              }),
+            }
+          )
+        ).json();
+        console.log(result);
+
+        let ids = [];
+        let bundles = ['openshift', 'rhel', 'console'];
+
+        Object.values(data.event_type_id).forEach((item) => {
+          ids = [...ids, ...Object.keys(item)];
+        });
```

Comment:
You can replace this with `map`
```suggestion
        const ids = Object.values(data.event_type_id).map((item) => Object.keys(item));
```

Reply:
this just returns null... I have tried to update it in a few ways but above is the only way i have been able to make it successful.

Reply:
Let's roll withe forEach then.